etcdserver/api/v3rpc/util.go

```diff
@@ -70,7 +68,7 @@ var toGRPCErrorMap = map[error]error{
 func togRPCError(err error) error {
 	grpcErr, ok := toGRPCErrorMap[err]
 	if !ok {
-		return status.Error(codes.Unknown, err.Error())
+		return err
```

Comment:
try to assert if this is gRPC status error, if it is then we should return the error directly or we should still use Unknown.

Reply:
> try to assert if this is gRPC status error

`context.DeadlineExceeded` is not `*status.statusError` type. So it will be passed with `codes.Unknown`?

Reply:
then we should make it status deadline exceed type.

Reply:
The status code is handled by gRPC server-side https://github.com/grpc/grpc-go/blob/551f29576314b126265b0f3bde41dc7e57c61e22/server.go#L877-L880.

So `context.DeadlineExceeded` will be converted to `codes.DeadlineExceeded` by gRPC.